File: FormalBench-data/diverse/rule_21/MinCoins.java

import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class MinCoins {

	public static int minCoins(int[] coins, int m, int v) {
		if (0 == v) {
			return 0;
		}

		if (v < 0) {
			return Integer.MAX_VALUE;
		}

		int minCoins = Integer.MAX_VALUE;

		for (int i = 0; i < m; i++) {
			int coin = coins[i];
			int remainingValue = v - coin;
			if (remainingValue >= 0) {
				int totalCoins = minCoins(coins, m, remainingValue);
				if (totalCoins != Integer.MAX_VALUE) {
					totalCoins += 1;
				}

				if (totalCoins < minCoins) {
					minCoins = totalCoins;
				}
			}
		}

		return minCoins;
	}
}

File: FormalBench-data/FilteredData/fault_mutants/run_ukcw__uc/MinCoins/mutants/19/c/MinCoins.c
#include "java_arrays.h"
#include <stdint.h>

static int32_t java_add(int32_t left, int32_t right) { return (int32_t)((uint32_t)left + (uint32_t)right); }
static int32_t java_sub(int32_t left, int32_t right) { return (int32_t)((uint32_t)left - (uint32_t)right); }

int32_t minCoins(JIntArray coins, int32_t m, int32_t v) {
    if ((v == INT32_C(0))) {
        return INT32_C(0);
    }
    if ((v < INT32_C(0))) {
        return INT32_MAX;
    }
    int32_t minCoinsValue = INT32_MAX;
    for (int32_t i = INT32_C(0); (i < m); i = java_add(i, INT32_C(1))) {
        int32_t coin = jarray_get(coins, i);
        int32_t remainingValue = java_sub(v, coin);
        if ((remainingValue >= INT32_C(0))) {
            int32_t totalCoins = minCoins(coins, m, remainingValue);
            if ((totalCoins < INT32_MAX)) {
                totalCoins = java_add(totalCoins, INT32_C(1));
            }
            if ((totalCoins < minCoinsValue)) {
                minCoinsValue = totalCoins;
            }
        }
    }
    return minCoinsValue;
}
